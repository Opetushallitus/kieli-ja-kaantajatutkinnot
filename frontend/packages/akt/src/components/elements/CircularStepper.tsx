import { CircularProgress, CircularProgressProps } from '@mui/material';

import { H3 } from 'components/elements/Text';
import { useCommonTranslation } from 'configs/i18n';
import { Color } from 'enums/app';

interface CircularStepperProps extends CircularProgressProps {
  value: number;
  phaseText: string;
  phaseDescription: string;
}

export const CircularStepper = ({
  phaseText,
  phaseDescription,
  ...rest
}: CircularStepperProps) => {
  const translateCommon = useCommonTranslation();
  const ariaLabel = `${translateCommon(
    'phase'
  )} ${phaseText}: ${phaseDescription}`;

  return (
    <div className="circular-stepper">
      <CircularProgress
        aria-label={ariaLabel}
        className="circular-stepper__progress"
        color={Color.Secondary}
        variant="determinate"
        {...rest}
      />
      <div className="circular-stepper__heading">
        <H3>{phaseText}</H3>
      </div>
    </div>
  );
};
