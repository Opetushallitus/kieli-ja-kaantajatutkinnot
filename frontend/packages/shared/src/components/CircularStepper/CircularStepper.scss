@use '../../styles/abstracts/variables';

.circular-stepper {
  display: inline-flex;
  position: relative;

  &__heading {
    align-items: center;
    bottom: 0;
    display: flex;
    justify-content: center;
    left: 0;
    position: absolute;
    right: 0;
    top: 0;
  }

  svg {
    background: radial-gradient(
      ellipse at center,
      variables.$color-primary 0%,
      variables.$color-primary 56%,
      variables.$color-primary-dark 59%,
      variables.$color-primary-dark 100%
    );
    border-radius: 50%;
  }
}
