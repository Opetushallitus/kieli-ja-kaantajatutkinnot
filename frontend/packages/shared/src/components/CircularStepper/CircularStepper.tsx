import { CircularProgress, CircularProgressProps } from '@mui/material';
import { visuallyHidden } from '@mui/utils';

import { Text } from '../../components';
import { Color } from '../../enums/common';
import { H3 } from '../Text/Text';

import './CircularStepper.scss';

interface CircularStepperProps extends CircularProgressProps {
  value: number;
  phaseText: string;
  ariaLabel: string;
}

export const CircularStepper = ({
  phaseText,
  ariaLabel,
  ...rest
}: CircularStepperProps) => {
  return (
    <div className="circular-stepper">
      <CircularProgress
        aria-label={ariaLabel}
        aria-hidden={true}
        className="circular-stepper__progress"
        color={Color.Secondary}
        variant="determinate"
        {...rest}
      />
      <div className="circular-stepper__heading" aria-hidden={true}>
        <H3>{phaseText}</H3>
      </div>
      <Text sx={visuallyHidden}>{ariaLabel}</Text>
    </div>
  );
};
