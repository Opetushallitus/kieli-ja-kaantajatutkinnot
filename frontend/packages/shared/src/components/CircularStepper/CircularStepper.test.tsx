import renderer from 'react-test-renderer';

import { CircularStepper } from './CircularStepper';

describe('CircularStepper', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <CircularStepper
          phaseText="Test Phase 1"
          ariaLabel="phase Test Phase 1: Description text"
          value={33}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
