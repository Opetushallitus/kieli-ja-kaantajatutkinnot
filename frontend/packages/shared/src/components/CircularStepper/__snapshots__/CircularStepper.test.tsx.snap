// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircularStepper should render correctly 1`] = `
<div
  className="circular-stepper"
>
  <span
    aria-hidden={true}
    aria-label="phase Test Phase 1: Description text"
    aria-valuenow={33}
    className="MuiCircularProgress-root MuiCircularProgress-determinate MuiCircularProgress-colorSecondary circular-stepper__progress css-1bx0sm6-MuiCircularProgress-root"
    role="progressbar"
    style={
      {
        "height": 40,
        "transform": "rotate(-90deg)",
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate css-oxts8u-MuiCircularProgress-circle"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={
          {
            "strokeDasharray": "126.920",
            "strokeDashoffset": "85.037px",
          }
        }
      />
    </svg>
  </span>
  <div
    aria-hidden={true}
    className="circular-stepper__heading"
  >
    <h3
      className="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
    >
      Test Phase 1
    </h3>
  </div>
  <p
    className="MuiTypography-root MuiTypography-body1 css-33u759-MuiTypography-root"
  >
    phase Test Phase 1: Description text
  </p>
</div>
`;
